src/flowgger/encoder/mod.rs

```diff
@@ -54,15 +56,27 @@ pub trait Encoder: CloneBoxedEncoder {
 }
 
 pub fn config_get_prepend_ts(config: &Config) -> Option<String> {
-    config
+    let prepend_ts = config
         .lookup("output.syslog_prepend_timestamp")
         .map_or(None, |bs| {
             Some(
                 bs.as_str()
                     .expect("output.syslog_prepend_timestamp should be a string")
                     .to_string(),
             )
-        })
+        });
+
+    if let Some(time_format) = prepend_ts {
+        let time_format = validate_time_format_input(
+            "syslog_prepend_timestamp",
+            &time_format,
+            SYSLOG_PREPEND_DEFAULT_TIME_FORMAT.to_string(),
+        );
+        Some(time_format)
+    } else {
+        // Not using syslog_prepend_timestamp - no need to validate
+        None
+    }
```

Comment:
i think that the more rusty way to do this is (and imho clearer)

```suggestion
    match prepend_ts {
        Some(time_format) => {
            let actual_time_format = validate_time_format_input(
                "syslog_prepend_timestamp",
                &time_format,
                SYSLOG_PREPEND_DEFAULT_TIME_FORMAT.to_string(),
            );
            Some(actual_time_format) 
        },
        // Not using syslog_prepend_timestamp - no need to validate
        None => None,
    }
```